src/server/detail/wrapped_json_path.h

```diff
@@ -79,9 +81,18 @@ struct CallbackResultOptions {
       : path_type(path_type_), saving_order(saving_order_), on_empty(on_empty_) {
   }
 
-  std::optional<JsonPathType> path_type = std::nullopt;
-  SavingOrder saving_order{SavingOrder::kSaveLast};
-  OnEmpty on_empty{OnEmpty::kSendWrongType};
+  void InitializeEmptyOptions(JsonPathType default_path_type, OnEmpty default_on_empty) {
```

Comment:
I find this confusing
calling this function will initialize the values if they are not already initialized? 
So do we have flows when we call this function twice with different values and only the first call should set the values?